include/drivers/gpio.h

```diff
@@ -450,6 +450,23 @@ struct gpio_callback {
  *
  * For internal use only, skip these in public documentation.
  */
+
+/* Used by driver api function pin_interrupt_configure, these are defined
+ * in terms of the public int flags so we can just mask and pass them
+ * through to the driver api
+ */
+enum gpio_int_mode {
+	GPIO_INT_MODE_DISABLED = GPIO_INT_DISABLE,
+	GPIO_INT_MODE_LEVEL = GPIO_INT_ENABLE,
+	GPIO_INT_MODE_EDGE = GPIO_INT_ENABLE | GPIO_INT_EDGE,
+};
+
+enum gpio_int_trig {
+	GPIO_INT_TRIG_LOW = GPIO_INT_LOW_0,
+	GPIO_INT_TRIG_HIGH = GPIO_INT_HIGH_1,
+	GPIO_INT_TRIG_BOTH = GPIO_INT_LOW_0 | GPIO_INT_HIGH_1,
```

Comment:
Let's add description of the trigger modes since one flag covers both a state and a transition.
```
	/* Trigger detection when input state is (or transitions to) physical low. */
	GPIO_INT_TRIG_LOW = GPIO_INT_LOW_0,
	/* Trigger detection when input state is (or transitions to) physical high. */
	GPIO_INT_TRIG_HIGH = GPIO_INT_HIGH_1,
	/* Trigger detection on pin rising or falling edge. */
	GPIO_INT_TRIG_BOTH = GPIO_INT_LOW_0 | GPIO_INT_HIGH_1,
```

Reply:
Updated.